src/main/java/com/example/bookstorewebapp/repository/cartitem/CartItemRepository.java

```diff
@@ -0,0 +1,25 @@
+package com.example.bookstorewebapp.repository.cartitem;
+
+import com.example.bookstorewebapp.model.CartItem;
+import jakarta.transaction.Transactional;
+import java.util.Set;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+public interface CartItemRepository extends JpaRepository<CartItem, Long>,
+        JpaSpecificationExecutor<CartItem> {
+    Set<CartItem> findAllByShoppingCartId(Long id);
+
+    @Modifying
+    @Transactional
```

Comment:
No reason to add Transactional here bacause you make one DB request by this method, I makes sence to add Transactional  to service methods that make several DB requests

Reply:
Agree I was overcomplicated.